packages/react-router/src/useLocation.ts

```diff
@@ -0,0 +1,10 @@
+import { useRouter } from "./useRouter";
+import type { ParsedLocation } from "./location";
+import type { FullSearchSchema } from "./routeInfo";
+import type { AnyRoute } from "./route";
+
+
+export function useLocation<TRouteTree extends AnyRoute = AnyRoute>(): ParsedLocation<FullSearchSchema<TRouteTree>> {
```

Comment:
Should we be supporting the select syntax here? Like

```tsx
const pathname = useLocation({ select: s => s.pathname })
```

Reply:
Yes probably